src/construction.cpp

```diff
@@ -820,17 +826,41 @@ void place_construction( const std::string &desc )
         cons.front()->explain_failure( pnt );
         return;
     }
-
+    std::list<item> used;
     const construction &con = *valid.find( pnt )->second;
-    g->u.assign_activity( activity_id( "ACT_BUILD" ), con.adjusted_time(), con.id );
+    // Set the trap that has the examine function
+    g->m.trap_set( pnt, tr_unfinished_construction );
+    // Use up the components
+    for( const auto &it : con.requirements->get_components() ) {
+        std::list<item> tmp = g->u.consume_items( it, 1, is_crafting_component );
+        used.splice( used.end(), tmp );
+    }
+    for( const auto &it : con.requirements->get_tools() ) {
+        g->u.consume_tools( it );
+    }
+    // Make the partial construction item
+    item partial_construction( "partial_construction", calendar::turn );
+    partial_construction.components = used;
+    // Add it to the world ( add_item will force placement on impassable furniture )
+    item &added_item = g->m.add_item( pnt, partial_construction );
+    // Add the vars
+    added_item.set_var( "time_passed", 0 );
+    added_item.set_var( "construction_time", con.time );
+    added_item.item_counter = 0;
+    added_item.set_var( "index", static_cast<int>( con.id ) );
+    std::string con_desc = string_format( _( "Unfinished task: %s, 0% complete" ), con.description );
+    added_item.set_var( "name", con_desc );
+    item_location item_loc( map_cursor( pnt ), &added_item );
+    g->u.assign_activity( activity_id( "ACT_BUILD" ), con.time, con.id );
```

Comment:
You don't need to set a time for the activity here since the duration is handled manually in `build_do_turn()`